Leanplum/Leanplum.m

```diff
@@ -551,6 +553,17 @@ + (void)startWithUserId:(NSString *)userId
         }];
     }];
     
+    //Batch calls every 15 mins.
+    [NSTimer scheduledTimerWithTimeInterval:HEARTBEAT_INTERVAL repeats:YES block:^(NSTimer * _Nonnull timer) {
```

Comment:
should we keep a reference to this timer? what happens when the app goes in background?

Reply:
let me check

Reply:
I tested this with a sample app on what happens if the app goes in background and then foreground again, will the timer still be resumed. It worked.  Also, checked with the existing sdk, we do not need to keep a reference, it might be more of a problem to keep a reference, to manage it. This one is pretty self contained.

So, we do not need to keep a reference. The timer will resume once the app comes back from background. And if the app gets killed. The start will be called and the timer will be kick started again.

Reply:
ok, as long as this isn't causing any crashes!